app/scripts/metamask-controller.js

```diff
@@ -93,7 +94,7 @@ module.exports = class MetamaskController {
 
     function logger (err, request, response) {
       if (err) return console.error(err)
-      if (!request.isMetamaskInternal) {
+      if (developmentMode && !request.isMetamaskInternal) {
         console.log(`RPC (${originDomain}):`, request, '->', response)
         if (response.error) {
           console.error('Error in RPC response:\n', response.error)
```

Comment:
Logging errors actually doesn't need to be only in development, and I think is appropriate for production.